kevm-pyk/src/kevm_pyk/cli.py

```diff
@@ -155,6 +445,19 @@ def default() -> dict[str, Any]:
             'counterexample_info': False,
         }
 
+    @staticmethod
+    def from_option_string() -> dict[str, str]:
+        return {
+            'counterexample-information': 'counterexample_info',
+            'no-counterexample-information': 'no_counterexample_info',
+            'node': 'nodes',
+            'node-delta': 'node_deltas',
+        }
+
+    @staticmethod
+    def get_argument_type() -> dict[str, Callable]:
+        return {'node': node_id_like, 'node-delta': arg_pair_of(node_id_like, node_id_like)}
```

Comment:
I think this is still a type mismatch. `KCFGShowOptions.nodes` is expected to be a list of node ids, but the TOML reading system expects a single node ID, and will assign `KCFGShowOptions.nodes` to that `int`. With the `get_argument_type` system you added maybe you can just set the argument type to `list`. This should apply to all the options where the `add_argument` call for that option has `action='append'`. That setting makes it so that the typing function specified in `add_argument`'s `type` parameter no longer matches the type that we want to end up as a field in the options class.